tutorials/week4-react.md

```diff
@@ -0,0 +1,511 @@
+-   [Understanding a React App](#react-component)
+    -   [Components](#react-component)
+    -   [Template For Structure of Function Component](#template-for-structure-of-function-component)
+    -   [Props](#props)
+    -   [State](#state)
+    -   [Communicating between Components](#communication-between-components)
+        - [Parent to Child](#parent-to-child-component-communication)
+        - [Child to Parent](#child-to-parent-component-communication)
+    -   [Handling Events](#handling-events)
+-   [React Hooks](#react-hooks)
+    -   [UseState](#usestate)
+    -   [UseEffect](#useeffect)
+
+## Creating a New React App
+
+Let's use **npx** and **create-react-app** to create a new React project
+
+-   **npx** stands for Node Package Execute. It is a runner that can execute any package that you want from the npm registry without even installing that package. In this case npx tool temporarily installs the create-react-app npm package and uses it to create our project.
+
+-   **create-react-app** is a command-line tool that we can use to quickly create a React and TypeScript app with lots of useful pieces.
+
+1. We use the create-react-app npm package to create a React and TypeScript project by entering the following:
+
+    ```ts
+    npx create-react-app my-app --template typescript
+    ```
+
+    - Note 1: This will create a new project directory called my-app under the current directory.
+    - Note 2: This will create a git repo in my-app, so you probably shouldn't do this inside a pre-existing git repository.
+
+2. Navigate to the project directory using the command:
+    ```ts
+    cd my-app
+    ```
+3. To start the development server for React, run the command:
+    ```ts
+    npm start
+    ```
+4. Navigate to http://localhost:3000/ to see the default react page.
+    - After a few seconds, a browser window opens, with our app running:
+      ![image](./assets/week4-react/start-screen.png)
+
+## React Component
+React follows a Component based architecture. A component is a
+repeatable html element with built-in state, business logic, and a
+lifecycle. The component may be something as simple as a single html
+element such as an input box, or a button, or a complex entity made up
+of other components.
+
+Components are the basic building blocks of a React application and they allow the developer to split the UI into independent and reusable
+pieces where each piece can be used in isolation. 
+### Creating A Function Component in React
+
+<a href="https://codesandbox.io/s/nervous-morse-o3pwqm?file=/src/tutorial/CreateFunctionComponent.tsx" target="_blank">view in sandbox</a>
+
+The simplest method to define a component is to write a function in Javascript. These components are also widely referred as functional stateless components but in the recent versions of React, they have the capabilities to contain state as well.
+```ts
+import * as React from "react";
+
+interface Props = {name?: string};
```

Comment:
Syntax error.

It should be: 

```
export interface Props {
  name?: string;
}
```

1.  No '='.
2. Also linting will convert it into two lines.

Also, can we add export? 
Although its very trivial, but I guess it will hint students that we can export interface also.